main/spprintf.c

```diff
@@ -306,8 +306,8 @@ static void xbuf_format_converter(void *xbuf, zend_bool is_char, const char *fmt
 					} else if (*fmt == '*') {
 						precision = va_arg(ap, int);
 						fmt++;
-						if (precision < 0)
-							precision = 0;
+						if (precision < -1)
+							precision = -1;
```

Comment:
I don't think it's quite as simple as that. In particular precision is used for more than just floating point numbers. E.g. if you do something like `("%.*s", -1, str)` it would try to use -1 as the string length. I think it would be best to drop the validation here and instead perform for the individual format specifiers.

Furthermore this is just spprintf -- we have similar code in snprintf as well and maybe a few other places.